src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
 public class WorkWithFile {
+
+    public static final String SUPPLY_CODE = "supply";
+    public static final String BUY_CODE = "buy";
+    public static final String RESULT = "result";
+    public static final int CODE_POSITION = 0;
+    public static final int VALUE_POSITION = 1;
+    public static final String DELIMITER = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> inputLines = getFileContents(fromFileName);
+
+        int buyBalance = 0;
+        int supplyBalance = 0;
+        for (String line: inputLines) {
+            String[] columns = line.split(DELIMITER);
+            if (BUY_CODE.equalsIgnoreCase(columns[CODE_POSITION])) {
+                buyBalance += Integer.parseInt(columns[VALUE_POSITION]);
+            } else if (SUPPLY_CODE.equalsIgnoreCase(columns[CODE_POSITION])) {
+                supplyBalance += Integer.parseInt(columns[VALUE_POSITION]);
+            }
+        }
```

Comment:
Consider dividing the logic.
For instance, this part of code can be moved to separate method -- createReport(), which will return a report which you'll write to the file.
![image](https://github.com/mate-academy/jv-work-with-file/assets/72043323/d136575f-598f-4ba2-93b5-94e5b3d0be92)

You will have a pretty code, divided in logical parts, such a(getStatisticMethod):
readFromFile
createReport(readLines)
writeToFile(report)

Reply:
Good catch.
Thank you!